src/sentry/region_to_control/consumer.py

```diff
@@ -0,0 +1,113 @@
+            },
+        )
+    )
+
+    processor = StreamProcessor(
+        consumer=consumer,
+        topic=Topic(settings.KAFKA_REGION_TO_CONTROL),
+        processor_factory=ProcessorFactory(
+            lambda commit: BatchProcessingStrategy(
+                commit,
+                worker=RegionToControlConsumerWorker(),
+                max_batch_size=max_batch_size,
+                max_batch_time=max_batch_time,
+            )
+        ),
+    )
+
+    def handler(*args) -> None:
+        processor.signal_shutdown()
+
+    signal.signal(signal.SIGINT, handler)
+    signal.signal(signal.SIGTERM, handler)
+
+    return processor
+
+
+class RegionToControlConsumerWorker(AbstractBatchWorker[KafkaPayload, RegionToControlMessage]):
+    def process_message(self, message: Message[KafkaPayload]) -> RegionToControlMessage:
+        raw = json.loads(message.payload.value.decode("utf8"))
+        return RegionToControlMessage.from_payload(raw)
+
+    def flush_batch(self, batch: Sequence[RegionToControlMessage]):
+        with metrics.timer("region_to_control.consumer.flush_batch"):
+            return self._flush_batch(batch)
+
+    def _flush_batch(self, batch: Sequence[RegionToControlMessage]):
+        for row in batch:
+            if row.user_ip_event:
+                updated, created = UserIP.objects.create_or_update(
+                    values=dict(
+                        user_id=row.user_ip_event.user_id,
+                        ip_address=row.user_ip_event.ip_address,
+                        last_seen=row.user_ip_event.last_seen,
+                        country_code=row.user_ip_event.country_code,
+                        region_code=row.user_ip_event.region_code,
+                    )
+                )
+
+                if created:
+                    metrics.incr("region_to_control.consumer.user_ip_event.created")
+                elif updated:
+                    metrics.incr("region_to_control.consumer.user_ip_event.updated", amount=updated)
+                else:
+                    # This happens when there is an integrity error adding the UserIP -- such as when user is deleted,
+                    # or the ip address does not match the db validation.  This is expected and not an error condition
+                    # in low quantities.
+                    metrics.incr("region_to_control.consumer.user_ip_event.stale_event")
+
+
+ProcessorT = TypeVar("ProcessorT", bound=ProcessingStrategy[KafkaPayload])
+Commit = Callable[[Mapping[Partition, Position]], None]
+
+
+class ProcessorFactory(ProcessingStrategyFactory[KafkaPayload], Generic[ProcessorT]):
+    """
+    Generic processor factory that defers to a callable.
+    """
+
+    constructor: Callable[[Commit], ProcessorT]
+
+    def __init__(self, constructor: Callable[[Commit], ProcessorT]):
+        self.constructor = constructor
+
+    def create_with_partitions(
+        self,
+        commit: Callable[[Mapping[Partition, Position]], None],
```

Comment:
```suggestion
        commit: Commit,
```

Should this be using the typevar defined earlier?

Reply:
Yes